src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,106 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double GROW_FACTOR = 1.5;
+    private T[] defaultArray;
```

Comment:
i don't think this name is clear enough, why is it default? it's used throughout the whole lifecycle of arraylist instance, it's only default when initialized. Including data type in the name is also not needed, Java is statically typed anyway
```suggestion
    private T[] elements;
```
```suggestion
    private T[] values;
```

Reply:
I do it for my personal understanding, this name is better for me, but I will change it for you anyway